src/elements/List/List.js

```diff
@@ -1,32 +1,102 @@
-import React, { Component, PropTypes } from 'react'
-import classNames from 'classnames'
+import cx from 'classnames'
+import React, { PropTypes } from 'react'
 import META from '../../utils/Meta'
+import * as sui from '../../utils/semanticUtils'
+import {
+  getUnhandledProps,
+  useValueAndKey,
+  useKeyOrValueAndKey,
+  useKeyOnly,
+} from '../../utils/propUtils'
 import ListItem from './ListItem'
 
-export default class List extends Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-  }
-
-  static _meta = {
-    library: META.library.semanticUI,
-    name: 'List',
-    type: META.type.element,
-  }
-
-  static Item = ListItem
-
-  render() {
-    const classes = classNames(
-      'ui',
-      this.props.className,
-      'list'
-    )
-    return (
-      <div {...this.props} className={classes}>
-        {this.props.children}
-      </div>
-    )
-  }
+function List(props) {
+  const {
+    size, aligned, bulleted, link, ordered, animated, celled, divided,
+    horizontal, inverted, relaxed, selection, children, className,
+    } = props
+
+  const classes = cx(
+    'ui',
+    size,
+    useValueAndKey(aligned, 'aligned'),
+    useKeyOnly(bulleted, 'bulleted'),
+    useKeyOnly(link, 'link'),
```

Comment:
For the latest on component augmentation (`component`/`elementType`/`as` prop) see https://github.com/TechnologyAdvice/stardust/issues/403.

For more on the `link` prop and rendering as an `a` tag see https://github.com/TechnologyAdvice/stardust/pull/382#discussion_r75187539.